### PR TITLE
I71 improve autcomplete

### PR DESCRIPTION
**Pull request by J-Son89**
Removed jquery ui autocomplete and instead using awesomplete as it is much lighter.
Some small styling changes too.

### CODE REVIEW

File: src/js/profile.js [hunk 1]
```diff
@@ -87,10 +87,24 @@ $(function() {
...
+            },
+
+            replace: function(text) {
+                var before = this.input.value.match(/^.+,\s*|/)[0];
+                this.input.value = before + text;
+            }
         });
+
+        // $("#input0").autocomplete({
+        //     source: friends,
```
Comment (smeetz) [inline src/js/profile.js]: Remove this and other .old code

File: src/js/profile.js [hunk 2]
```diff
@@ -87,10 +87,24 @@ $(function() {
...
 
-        $("#input0").autocomplete({
-            source: friends,
-            appendTo: "#modal"
+        $("#input0").attr("data-list", friends );
+
+        new Awesomplete( ("#input0"), {
+            
+            filter: function(text, input) {
+                return Awesomplete.FILTER_CONTAINS(text, input.match(/[^,]*$/)[0]);
```
Comment (smeetz) [inline src/js/profile.js]: Could you add a short notice about these regexes?

File: src/less/all.less [hunk 3]
```diff
@@ -14,4 +14,5 @@
 @import 'src/less/workingfiles/modal.less';
 @import 'src/less/workingfiles/media.less';
 @import 'src/less/workingfiles/confirmpopup.less';
+@import (less) 'src/less/workingfiles/awesomplete.css';
```
Comment (smeetz) [inline src/less/all.less]: Can you change `awesomplete.css` to `awesomplete.less`?

File: src/less/workingfiles/submit.less [hunk 4]
```diff
@@ -342,15 +295,21 @@ th {
...
     margin-top: 5px;
     width: 12rem; /*200px;*/
-    margin-left: -45px; /*45px;*/
+    //margin-left: -45px; /*45px;*/
 }
+.facet-additional-input::-webkit-scrollbar {
+        display: none;
+        z-index:1;
+        position: relative;
+        //not working for mozilla yet
```
Comment (smeetz) [inline src/less/workingfiles/submit.less]: After a quick google Firefox seems rekt, do you how IE, Edge or Safari handles this stuff?

**Comment by smeetz** [comment]
[Replace jquery UI autocomplete](https://trello.com/c/VQxz7oCQ/116-replace-jquery-ui-autocomplete)

File: src/js/profile.js (hunk 1 above)
Comment (J-Son89) [inline src/js/profile.js]: Yup :P

File: src/less/workingfiles/submit.less (hunk 4 above)
Comment (J-Son89) [inline src/less/workingfiles/submit.less]: This was to remove the inner scroll bar on the autocomplete window. I'm not sure it's necessary so I will just remove this for now.

File: src/js/profile.js (hunk 2 above)
Comment (J-Son89) [inline src/js/profile.js]: adding to collection/ users and will update the code now.
This short notice should be commented in the code file or on github? or both?

Comment (smeetz) [inline src/js/profile.js]: A comment in the source code is good enough!

File: src/less/workingfiles/submit.less (hunk 4 above)
Comment (smeetz) [inline src/less/workingfiles/submit.less]: Yeah it seems like only webkit supports that feature atm so unless it's really :sparkles: then we can skip it imo

File: src/js/profile.js (hunk 2 above)
Comment (J-Son89) [inline src/js/profile.js]: I see what you mean about the self.email part. Could we not set the default being queried is the users email unless specified?
Everything else has worked fine afaik

Comment (smeetz) [inline src/js/profile.js]: Yeah that's a good compromise:

```
/friends --> friends of logged in user
/friends?email=... --> friends of queried user
```

Comment (J-Son89) [inline src/js/profile.js]: sounds goood to me

**Comment by J-Son89** [comment]
Oh yeah, I'm pretty sure jquery-ui isn't used so I'll take care of that.

**Pull request merged**